MODULE.bazel

```diff
@@ -0,0 +1,46 @@
+JDK_VERSIONS = ["11", "15", "16", "17"]
+PLATFORMS = ["linux", "macos", "macos_aarch64", "win"]
+
+# Remote JDK repos for those Linux platforms are only defined for JDK 11.
+EXTRA_REMOTE_JDK11_REPOS = [
+    "remotejdk11_linux_aarch64",
+    "remotejdk11_linux_ppc64le",
+    "remotejdk11_linux_s390x",
+]
+
+REMOTE_JDK_REPOS = [("remotejdk" + version + "_" + platform) for version in JDK_VERSIONS for platform in PLATFORMS] + EXTRA_REMOTE_JDK11_REPOS
+
+REMOTE_JAVA_TOOLCHAINS = [("@" + name + "_toolchain_config_repo//:toolchain") for name in REMOTE_JDK_REPOS]
+
+module(
+    name = "rules_java",
+    version = "4.0.0",
+    compatibility_level = 1,
+    toolchains_to_register = [
+        "//toolchains:all",
+        "@local_jdk//:runtime_toolchain_definition",
+    ] + REMOTE_JAVA_TOOLCHAINS,
+)
+
+bazel_dep(name = "bazel_skylib", version = "1.0.3")
+bazel_dep(name = "platforms", version = "0.0.4")
+bazel_dep(name = "rules_cc", version = "0.0.1")
+bazel_dep(name = "rules_proto", version = "4.0.0")
```

Comment:
rules_proto also don't seem to be needed.

Reply:
rules_proto is used in `remote_java_tools` (See [here](https://github.com/bazelbuild/bazel/blob/28c09f29bc357a4b5f4c4fd593e60cef5fa08d27/tools/jdk/BUILD.java_tools#L8)), because currently I'm loading java tools via module extension instead of turning it into a separate module, I'll have to declare its dependencies in rules_java as well.

Do you think java tools deserve its own module, if so I can do it in a follow up change. Added a comment here for now.

Reply:
TL;DR I don't think java_tools needs a bzlmod module.

I think the sources  that generate java_tools should be moved from bazel github repo into rules_java github repo (this would allow easier version tracking). Then the tests should be moved as well and the release procedure for rules_java should be able to build and pack java_tools (for three archs that we have) into one big release.

Reply:
Then the java_tools repo should be deprecated, right? Sounds good to me.